rfcs/2023-01-06-layout.md

```diff
@@ -0,0 +1,461 @@
+- Flex gap has ~91% global [coverage](https://caniuse.com/?search=flex%20gap) (support added in some browsers mid 2020). Although the support looks decent, this might be problematic for some products supporting these browsers. There is a [post CSS plugin](https://github.com/gavinmcfarland/flex-gap-polyfill) which can polyfill this _(with some known limitations)_. Alternative is to internally translate gap properties to utilize CSS selectors such as [lobotomized owl](https://alistapart.com/article/axiomatic-css-and-lobotomized-owls/) `* + *`.
+- Going forward with native `gap` supports seems like a fine choice here in favor of keeping things simple and native. Tradeoff being, older browsers will be able to render the layout with a degraded experience _(without spacing)_.
+- CSS grid has ~96% global [coverage](https://caniuse.com/?search=css%20grid). Although we can only natively support it on web, it's worth the investment since it can simplify building grid based layouts.
+
+**Few points to note for React Native**
+
+- React Native [recently added](https://reactnative.dev/blog/2023/01/12/version-071) native support for `gap` specific properties in `v0.71`.
+- Polyfilling this on native for older releases seems bit hacky _(with cloning and wrapping child elements)_ and unneeded because unlike browsers, all users can receive an updated React Native v0.71 app, tradeoff being we will only be supporting `gap` based props in React Native going forward from `0.71`, so consumer apps should update their React Native version in order to consume these APIs.
+- No support for grid props since it isn't supported natively.
+- The TypeScript typings and autosuggestions should appropriately be adjusted for web and native to accommodate the differences, eg. grid properties, `inline-flex`, `block` display properties won't be supported. This can be done with [`moduleSuffixes`](https://www.typescriptlang.org/tsconfig#moduleSuffixes) detailed in the [related RFC](./writing-cross-platform-typescript.md).
+
+We have an [internal Box component](../packages/blade/src/components/Box/) but it exposes lot more properties than the ones associated with layout. It seems like a good idea to have a different public `Box` component with the restrictive but fully covered layout properties from above table.
+
+#### `Flex` component
+
+A helper component over the `Box` component with `display` set as `Flex`. Arguably, this component seems unneeded without flex specific shorthand properties because the behavior can be achieved with a `display="flex"` prop on `Box` but it improves code readability. Supports same properties.
+
+#### `Grid` component
+
+A helper **web only** component over the `Box` component with `display` set as `grid`. Similar to `Flex` component, it may be argued this component is unneeded with `display="grid"` set on `Box` but it improves code readability. Supports same properties.
+
+#### `Spacer` component
```

Comment:
if we plan to expose layout props(like `sx`) on all of our elements then do we see a need for this component?